src/easylink/implementation.py

```diff
@@ -106,23 +161,50 @@ def __init__(
         output_slots: Iterable["OutputSlot"] = (),
     ):
         self.name = name
+        """The name of this NullImplementation."""
         self.input_slots = {slot.name: slot for slot in input_slots}
+        """A mapping of :class:`InputSlots<easylink.graph_components.InputSlot>`
+        names to their instances."""
         self.output_slots = {slot.name: slot for slot in output_slots}
+        """A mapping of :class:`OutputSlots<easylink.graph_components.OutputSlot>`
+        names to their instances."""
         self.schema_steps = [self.name]
+        """The requested :class:`~easylink.pipeline_schema.PipelineSchema`
+        :class:`~easylink.step.Step` names for which this NullImplementation is
+        expected to be responsible."""
         self.combined_name = None
+        """The name of the combined implementation that this NullImplementation 
+        is part of. This is definitionally None for a NullImplementation."""
 
 
 class PartialImplementation:
-    """A representation of one part of a combined implementation that spans multiple steps.
-
-    A PartialImplementation is what is initially added to the implementation graph when
-    a combined implementation is used (i.e. an implementation that spans multiple steps).
-    We initially add a node for _each_ step, which has as its "implementation" attribute a
-    PartialImplementation. Such a graph is not yet fit to run. When we make our second
-    pass through, after the flat (non-hierarchical) PipelineGraph has been created, we find the set of
-    PartialImplementation nodes corresponding to each combined implementation and
-    replace them with a single node with a true Implementation representing the combined
-    implementation.
+    """A representation of one part of a combined implementation that spans multiple :class:`Steps<easylink.step.Step>`.
+
+    A PartialImplementation is what is initially added to the :class:`~easylink.graph_components.ImplementationGraph`
+    when a so-called "combined implementation" is used (i.e. an :class:`Implementation`
```

Comment:
We should discuss as a team and determine the best approach to take regarding adding links in these docstrings. While having everything link like this is great, it's also super annoying to do (and also prone to typos). Some options:
- just don't link anything
- Within a given class, link the first mention and then stop linking thereafter. 
- Don't link anything inline, but add a `See Also` section with the links.

Reply:
Let's talk about this in mid-day if there's nothing else to discuss. I really like a lot of what you're doing here in your docs.

Reply:
I'd suggest linking the first mention only!